iceoryx_hoofs/test/moduletests/test_posix_mutex.cpp

```diff
@@ -34,7 +34,7 @@ class Mutex_test : public Test
     void SetUp() override
     {
         internal::CaptureStderr();
-        m_deadlockWatchdog.watchAndActOnFailure([] { std::terminate(); });
+        deadlockWatchdog.watchAndActOnFailure([] { std::terminate(); });
```

Comment:
assert just return and does not unblock the deadlock

Reply:
Right.We could in theory still do this in another way by essentially creating a NO_DEATH test which under normal conditions expects not to terminate but can terminated like here due to timeout. I did something like this some  time ago (not with timeout) in the C binding with an invalid options test case, Gtest has the necessary macro infrastructure. 

This would work (I think) because those are forked and if we terminate the forked test it would fail and a blocked mutex is of no concern since thre process is terminated anyway.

But it is not worth it to do so, I just think it is possible in principle.